include/dmlc/data.h

```diff
@@ -73,6 +73,8 @@ class Row {
   const real_t *label;
   /*! \brief weight of the instance */
   const real_t *weight;
+  /*! \brief session-id of the instance */
+  const size_t *qid;
```

Comment:
use int32_t, as size_t is not portable across platforms

Reply:
if there is worry about possible integer overflow, use uint64_t instead

Reply:
thanks, I have changed it to uint64_t.